osclib/stagingapi.py

```diff
@@ -1072,25 +1072,27 @@ def map_ring_package_to_subject(self, project, pkg):
 
         return project
 
-    def get_sub_packages(self, pkg, project=None):
+    def get_sub_packages(self, package, project=None):
         """
         Returns a list of packages that need to be linked into rings
         too. A package is actually a tuple of project and package name
         """
         ret = []
         if not project:
-            project = self.ring_packages.get(pkg)
+            project = self.ring_packages.get(package)
         if not project:
             return ret
-        url = self.makeurl(['source', project, pkg],
+        url = self.makeurl(['source', project, package],
                            {'cmd': 'showlinked'})
 
         # showlinked is a POST for rather bizzare reasons
         f = http_POST(url)
         root = ET.parse(f).getroot()
 
         for pkg in root.findall('package'):
-            ret.append((pkg.get('project'), pkg.get('name')))
+            # ensure sub-package is valid
+            if pkg.get('project') in self.rings and pkg.get('name') != package:
```

Comment:
1. sure, but we want to make it work for `adi` so why make work for ourselves by needlessly introducing more code that will have to be changed?
2. I agree that is a subset of what is valid. In factory right now there are packages with linked subpackages that are not in rings. `adi` code will need to look at the packages in target project to get a list. This same code will need to work in that case. A valid link should be to the something that starts with the target project.

My point was not that this change by itself will cause issues, but why write code that will cause us issues down the road.